### PR TITLE
Update BulkData 2.0.0: Update the OperationDefinitions and CapabilityStatement #3082 

### PR DESCRIPTION
**Pull request by prb112**
- Optimization to keep from establishing unnecessary Partitions. 
- Change the Defaults to match the actual fhir-server defaults.

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java [hunk 1]
```diff
@@ -61,13 +74,46 @@ public PartitionPlan mapPartitions() throws Exception {
         ConfigurationAdapter adapter = ConfigurationFactory.getInstance();
         adapter.registerRequestContext(ctx.getTenantId(), ctx.getDatastoreId(), ctx.getIncomingUrl());
 
+        // Check resourceType needs to be processed
```
Comment (lmsurpre) [inline fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java:77]: smart

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java [hunk 2]
```diff
@@ -61,13 +74,46 @@ public PartitionPlan mapPartitions() throws Exception {
...
+        List<String> target = new ArrayList<>();
+        SystemExportResourceHandler handler = new SystemExportResourceHandler();
+        for (String resourceType : resourceTypes) {
+            FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper(handler.getSearchHelper());
+            FHIRPersistence fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
+            Map<String, List<String>> queryParameters = new HashMap<>();
+            Class<? extends Resource> rt = ModelSupport.getResourceType(resourceType);
+            FHIRSearchContext searchContext = handler.getSearchHelper().parseQueryParameters(rt, queryParameters);
+            searchContext.setPageSize(1);
+            searchContext.setPageNumber(1);
```
Comment (lmsurpre) [inline fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java:87]: would be a little faster if you skip the count (`_total=none`).  that count query can be expensive if there are no search parameters.

Comment (prb112) [reply fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java]: I switched to the changes API. It's lightly more effective here.

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/system/SystemExportPartitionMapper.java [hunk 3]
```diff
@@ -51,16 +65,53 @@ public PartitionPlan mapPartitions() throws Exception {
...
+        List<String> target = new ArrayList<>();
+        SystemExportResourceHandler handler = new SystemExportResourceHandler();
+        for (String resourceType : resourceTypes) {
+            FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper(handler.getSearchHelper());
+            FHIRPersistence fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
+            Map<String, List<String>> queryParameters = new HashMap<>();
+            Class<? extends Resource> rt = ModelSupport.getResourceType(resourceType);
+            FHIRSearchContext searchContext = handler.getSearchHelper().parseQueryParameters(rt, queryParameters);
+            searchContext.setPageSize(1);
+            searchContext.setPageNumber(1);
```
[the remark this answers is not in the document]
Comment (prb112) [reply fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/system/SystemExportPartitionMapper.java]: Switched to changes API

File: fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java (hunk 1 above)
Comment (prb112) [reply fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java:79]: @lmsurpre if you can re-review, thanks, Paul

Comment (prb112) [reply fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/patient/PatientExportPartitionMapper.java:79]: Great - it should be a bit better

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**